CommonPackage.cmake

```diff
@@ -1,12 +1,23 @@
-# Copyright (c) 2014 Stefan.Eilemann@epfl.ch
+# Copyright (c) 2014-2015 Stefan.Eilemann@epfl.ch
+#                         Daniel.Nachbaur@epfl.ch
 
-# Provides common_package(Name args) which improves find_package.
-# First invokes find_package with all the given arguments, and then
-# falls back to using pkg_config if available. The pkg_config path
-# does only implement the version, REQUIRED and QUIET find_package
-# arguments (e.g. no COMPONENTS)
+# Provides common_package(Name args) and common_package_post() which improves
+# find_package.
+#
+# common_package()
+#  - QUIET if COMMON_PACKAGE_QUIET option is set
+#  - -isystem if SYSTEM argument is passed; for convenience, Boost is always SYSTEM
+#  - first tries find_package with all the given arguments, and then falls back
+#    to using pkg_config if available (no component support for pkg_config though)
+#  - processes ${PROJECT_SOURCE_DIR}/CMake/FindPackagesPost.cmake to tweak result
+#    of common_package_post()
+#  - sets include_directories() and link_directories() accordingly
+#
+# common_package_post()
+#  - generates defines${SYSTEM}.h and options.cmake for found packages
```

Comment:
Clarify the difference between defines.h and defines${SYSTEM}.h? I'm not sure why there are two files. I'm also not sure how options.cmake is meant to be used...